ingestion/src/metadata/ingestion/source/database/redshift/connection.py

```diff
@@ -40,4 +49,46 @@ def test_connection(engine: Engine) -> None:
     """
     Test connection
     """
-    test_connection_db_common(engine)
+
+    def custom_executor(engine, statement):
+        cursor = engine.execute(statement)
+        return list(cursor.all())
+
+    inspector = inspect(engine)
+    steps = [
+        TestConnectionStep(
+            function=partial(
+                custom_executor,
+                statement="SELECT datname FROM pg_database",
```

Comment:
let's move to queries.py

Reply:
and update it from https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/source/database/redshift/metadata.py#L434